actionpack/lib/action_controller/metal/flash.rb

```diff
@@ -33,10 +33,12 @@ def add_flash_types(*types)
         types.each do |type|
           next if _flash_types.include?(type)
 
-          define_method(type) do
-            request.flash[type]
+          if respond_to? :helper_method
+            define_method(type) do
+              request.flash[type]
+            end
+            helper_method type
           end
-          helper_method type
```

Comment:
We can execute
```ruby
define_method(type) do
  request.flash[type]
end
```
unless `#helper_method` defined,

so probably fix should be like:
```ruby
helper_method type if respond_to? :helper_method
```

Reply:
I'll send patch for that.